lib/domain/usecase/room/chat_room_search_interactor.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/config/app_config.dart';
+import 'package:fluffychat/domain/app_state/room/chat_room_search_state.dart';
+import 'package:fluffychat/utils/string_extension.dart';
+import 'package:matrix/matrix.dart';
+
+class ChatRoomSearchInteractor {
+  Stream<Either<Failure, Success>> execute({
+    required Timeline timeline,
+    required String keyword,
+    required int currentIndex,
```

Comment:
what is it? Index of what?

Reply:
updated to currentEventIndex

Reply:
you can read the ADR to make clear about search logic
- Employ Dart for finding the index of the first matching message.
- When the user presses next/previous, search for the index of the next matching message.